Formula/xinit.rb

```diff
@@ -0,0 +1,152 @@
+          #{opt_libexec}
+          #{opt_libexec}/privileged_startx
+          #{opt_prefix}
+          #{opt_bin}
+          #{HOMEBREW_PREFIX}/var/homebrew/linked/xinit
+        Then bootstrap the service:
+          `sudo launchctl bootstrap system /Library/LaunchDaemons/#{privileged_plist}`
+        This will require manual removal of these paths using `sudo rm` on
+        brew upgrade/reinstall/uninstall.
+      EOS
+    end
+  end
+
+  resource "font_cache" do
+    url "https://raw.githubusercontent.com/XQuartz/XQuartz/XQuartz-2.8.1/base/opt/X11/bin/font_cache"
+    sha256 "8f27cf55e5053686350fce6ea1078e314bd5b2d752e9da1b9051541e643e79d6"
+  end
+
+  resource "98-user.sh" do
+    url "https://raw.githubusercontent.com/XQuartz/XQuartz/XQuartz-2.8.1/base/opt/X11/etc/X11/xinit/xinitrc.d/98-user.sh"
+    sha256 "b417aea949931b7c03133535c3b5146b9403b8c3482a1c1d0a5dc01c07876c84"
+  end
+
+  def install
+    configure_args = std_configure_args + %W[
+      --with-bundle-id-prefix=homebrew.mxcl
+      --with-launchagents-dir=#{prefix}
+      --with-launchdaemons-dir=#{prefix}
+      --with-xserver=#{Formula["xserver"].opt_bin}/X
+    ]
+
+    system "./configure", *configure_args
+    system "make", "RAWCPP=tradcpp"
+    system "make", "install"
+
+    # Set `bindir` to `HOMEBREW_PREFIX/bin` so `X11.app` can find `xauth`
+    # and other utilities, otherwise `X11.app` will restart infinitely when
+    # launching from Finder.
+    inreplace bin/"startx", /(?<=^bindir=).*/, HOMEBREW_PREFIX/"bin"
+    # install scripts
+    if OS.mac?
+      bin.install resource("font_cache")
+      inreplace bin/"font_cache" do |s|
+        # set `X11DIR`
+        s.gsub! "/opt/X11", HOMEBREW_PREFIX
+        # provided by formula `procmail`
+        s.gsub! "/usr/bin/lockfile", HOMEBREW_PREFIX/"bin/lockfile"
+        # set `X11FONTDIR`, align with formula `font-util`
+        s.gsub! "/share/fonts", "/share/fonts/X11"
+      end
+
+      (prefix/"etc/X11/xinit/xinitrc.d").install resource("98-user.sh")
+
+      # align with formula `font-util`
+      font_paths = %w[misc TTF OTF Type1 75dpi 100dpi system_fonts].map do |f|
+        HOMEBREW_PREFIX/"share/fonts/X11"/f
+      end
+      font_paths += %w[$HOME/.fonts $HOME/Library/Fonts /Library/Fonts]
+      font_paths.map! do |p|
+        %Q([ -e #{p}/fonts.dir ] && fontpath="$fontpath,#{p}#{",#{p}/:unscaled" if /\d+dpi/.match? p}")
+      end
+      (prefix/"etc/X11/xinit/xinitrc.d/10-fontdir.sh").write <<~SH
+        #!/bin/sh
+        if [ -x #{HOMEBREW_PREFIX}/bin/xset ] ; then
+          fontpath="built-ins"
+          #{font_paths.join "\n  "}
+
+          #{HOMEBREW_PREFIX}/bin/xset fp= "$fontpath"
+          unset fontpath
+        fi
+      SH
+
+      (prefix/"etc/X11/xinit/xinitrc.d/99-quartz-wm.sh").write <<~SH
+        #!/bin/sh
+        [ -n "${USERWM}" -a -x "${USERWM}" ] && exec "${USERWM}"
+        [ -x #{HOMEBREW_PREFIX}/bin/quartz-wm ] && exec #{HOMEBREW_PREFIX}/bin/quartz-wm
```

Comment:
```suggestion
        [ -x #{HOMEBREW_PREFIX}/bin/quartz-wm ] && exec #{HOMEBREW_PREFIX}/bin/quartz-wm
        [ -x /usr/X11/bin/quartz-wm ] && exec /usr/X11/bin/quartz-wm
```
Should we support reusing `quartz-wm` from `XQuartz`？